daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala

```diff
@@ -220,23 +235,37 @@ private[lf] final class CommandPreprocessor(
     }
 
   @throws[Error.Preprocessing.Error]
-  def unsafePreprocessApiCommands(cmds: ImmArray[command.ApiCommand]): ImmArray[speedy.Command] = {
+  def unsafePreprocessApiCommands(cmds: ImmArray[command.ApiCommand]): ImmArray[speedy.Command] =
+    preprocessImmArray[command.ApiCommand, speedy.Command](cmds, unsafePreprocessApiCommand)
+
+  @throws[Error.Preprocessing.Error]
+  def unsafePreprocessDisclosedContracts(
+      discs: ImmArray[command.DisclosedContract]
+  ): ImmArray[speedy.DisclosedContract] =
+    preprocessImmArray[command.DisclosedContract, speedy.DisclosedContract](
+      discs,
+      unsafePreprocessDisclosedContract,
+    )
 
+  private[preprocessing] def preprocessImmArray[A, B](
```

Comment:
I should have removed this wart in #10620, could you :fire: it here ?

Reply:
ah cool, I thought there was some importance to not using `map`.